src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/CompiledXPathExpr.cs

```diff
@@ -141,6 +143,7 @@ public override IXsltContextVariable ResolveVariable(string prefix, string name)
             {
                 throw XPathException.Create(SR.Xp_UndefinedXsltContext);
             }
+            [RequiresUnreferencedCode(Scripts.ExtensionFunctionCannotBeStaticallyAnalyzed)]
```

Comment:
I think that ResolveFunction as a concept is not trim problematic - virtual methods are almost always like that.
The problem is the specific implementation in `XsltCompileContext`.

I think it would be better to leave this method unannotated and suppress warnings in the `XsltCompileContext` version of it.

It would also avoid suppressions which rely on complex interactions of several types (like the one in FunctionQuery).